### ISSUE TITLE
Hitting Ghast fireballs make the game crash

### ISSUE THREAD

**Issue by gamerjkt** [apace100/origins-fabric#210]
The title says it all

**Comment by minitolgy** [comment]
Same issue, I'm giving my crash-log from that.

```
---- Minecraft Crash Report ----

Time: 17.3.21 21:36
Description: Unexpected error

java.lang.ClassCastException: net.minecraft.class_1674 cannot be cast to net.minecraft.class_1309
	at Not Enough Crashes deobfuscated stack trace.(1.16.5+build.5)
	at net.minecraft.entity.player.PlayerEntity.md23bcbc$lambda$modifyDamage$0$9(PlayerEntity:20877)
	at net.minecraft.class_1657$$Lambda$10188/1425588777.test(Unknown Source)
	at io.github.apace100.origins.component.OriginComponent.lambda$modify$2(OriginComponent.java:84)
	at io.github.apace100.origins.component.OriginComponent$$Lambda$9847/1002217469.test(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.LinkedList$LLSpliterator.forEachRemaining(LinkedList.java:1235)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at io.github.apace100.origins.component.OriginComponent.modify(OriginComponent.java:85)
	at io.github.apace100.origins.component.OriginComponent.modify(OriginComponent.java:74)
	at net.minecraft.entity.player.PlayerEntity.localvar$dpi000$modifyDamage(PlayerEntity:20877)
	at net.minecraft.entity.player.PlayerEntity.attack(PlayerEntity:1166)
	at net.minecraft.client.network.ClientPlayerInteractionManager.attackEntity(ClientPlayerInteractionManager:371)
	at net.minecraft.client.MinecraftClient.doAttack(MinecraftClient:1389)
	at net.minecraft.client.MinecraftClient.handleInputEvents(MinecraftClient:1706)
	at net.minecraft.client.MinecraftClient.tick(MinecraftClient:1534)
	at net.minecraft.client.MinecraftClient.render(MinecraftClient:1021)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:681)
	at net.minecraft.client.MinecraftClient.handler$bej000$afterCrashHandled(MinecraftClient:8904)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:704)
	at net.minecraft.client.main.Main.main(Main:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Affected level --
Details:
	All players: 1 total; [class_746['ZombieTolgy'/3737, l='ClientLevel', x=206.99, y=59.03, z=-53.30]]
	Chunk stats: SodiumChunkCache: 510
	Level dimension: minecraft:the_nether
	Level spawn location: World: (144,67,-254), Chunk: (at 0,4,2 in 9,-16; contains blocks 144,0,-256 to 159,255,-241), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Level time: 26442 game time, 26442 day time
	Server brand: fabric
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:449)
	at net.minecraft.class_310.method_1587(class_310.java:2239)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:28)
	at net.minecraft.class_310.modify$bej000$onCrash(class_310.java:8912)
	at net.minecraft.class_310.method_1514(class_310.java:704)
	at net.minecraft.class_310.handler$bej000$afterCrashHandled(class_310.java:8904)
	at net.minecraft.class_310.method_1514(class_310.java:704)
	at net.minecraft.client.main.Main.main(Main.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 2048996872 bytes (1954 MB) / 3791650816 bytes (3616 MB) up to 6442450944 bytes (6144 MB)
	CPUs: 12
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx6G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Suspected Mods: Fabric Loader (fabricloader), Origins (origins)
	Fabric Mods: 
		adorn: Adorn 1.13.0+1.16.5
		advanced_runtime_resource_pack: Runtime Resource Pack 0.2.9
		advancementinfo: AdvancementInfo 1.16.5-fabric0.30.0-1.2
		adventurez: AdventureZ 1.2.0
		alaskanativecraft: Alaska Native Craft 1.1.1
		angerable-patch: Angerable Patch 1.1.0-1.16.4
		animal_feeding_trough: Animal Feeding Trough 1.0.0
		anycapes: AnyCapes 1.0.2
		appleskin: AppleSkin 1.0.11
		appliedenergistics2: Applied Energistics 2 8.2.0-alpha.3
		artifacts: Artifacts 2.3.0+fabric
		autoconfig1u: Auto Config v1 Updated 3.3.1
		bambootweaks: Bamboo Tweaks unspecified
		battletowers: Battle Towers 3.3.3-1.16.4
		bedspreads: Bedspreads 1.3.1-1.16.4
		bettercaves: YUNG's Better Caves 1.16.5-1.1
		betterend: Better End 0.9.1-pre
		bettergraves: Better Graves 0.4.1
		bettermineshafts: YUNG's Better Mineshafts 1.16.4-1.0.1
		betternether: Better Nether 5.0.7
		bitsandchisels: BitsAndChisels 2.1.2
		blue_endless_jankson: jankson 1.2.0
		cameraoverhaul: Camera Overhaul 1.2.2-fabric-universal
		campanion: Campanion 1.3.5
		can-i-mine-this-block: cAn i MiNe thIS bLOCk? 1.0.2
		cardinal-components: Cardinal Components API 2.7.13
		cardinal-components-base: Cardinal Components API (base) 2.7.13
		cardinal-components-block: Cardinal Components API (blocks) 2.7.13
		cardinal-components-chunk: Cardinal Components API (chunks) 2.7.13
		cardinal-components-entity: Cardinal Components API (entities) 2.7.13
		cardinal-components-item: Cardinal Components API (items) 2.7.13
		cardinal-components-level: Cardinal Components API (world saves) 2.7.13
		cardinal-components-scoreboard: Cardinal Components API (scoreboard) 2.7.13
		cardinal-components-util: Cardinal Components API (utilities) 2.7.13
		cardinal-components-world: Cardinal Components API (worlds) 2.7.13
		cavebiomes: Cave Biomes 0.6.2
		clickthrough: ClickThrough 1.16.5-fabric0.30.0-0.4
		clipboards: Clipboards 1.1.0+1.16.4
		cloth-armor-api-v1: Cloth Armor API v1 1.4.8
		cloth-basic-math: Cloth Basic Math 0.5.1
		cloth-client-events-v0: Cloth Client Events v0 1.5.47
		cloth-common-events-v1: Cloth Common Events v1 1.4.8
		cloth-config2: Cloth Config v4 4.11.14
		cloth-durability-bar-api-v1: Cloth Durability Bar API v1 1.4.8
		club_minnced_discord-webhooks: discord-webhooks 0.5.4
		com_fasterxml_jackson_core_jackson-annotations: jackson-annotations 2.10.1
		com_fasterxml_jackson_core_jackson-core: jackson-core 2.10.1
		com_fasterxml_jackson_core_jackson-databind: jackson-databind 2.10.1
		com_neovisionaries_nv-websocket-client: nv-websocket-client 2.10
		com_squareup_okhttp3_okhttp: okhttp 3.13.0
		com_squareup_okio_okio: okio 1.17.2
		com_vdurmont_emoji-java: emoji-java 5.1.1
		couplings: Couplings 1.3.3
		crimsonmoon: Crimson Moon 2.0.3-1.16.5
		croptopia: Croptopia 1.3.0
		crowdin-translate: CrowdinTranslate 1.2
		curios: Curios API 0.0.13-1.16.5
		dazzle: Dazzle 2 0.9.0
		disable_custom_worlds_advice: Disable Custom Worlds Advice 1.2
		dmnhwylaff: HWYLAFastFix 1.0.0
		dragonloot: DragonLoot 1.0.4
		dual_riders: DualRiders 1.0.5
		dungeonsmod: Dungeons Mod Lite 1.16.5-1.0.0
		dynamicsoundfilters: Dynamic Sound Filters 1.2.0+1.16.5
		earthtojavamobs: Earth2Java 1.6.1+1.16.4
		ecotones: Ecotones 0.7.2
		eldritch_mobs: Eldritch Mobs 1.6.8
		elementalcreepers: Elemental Creepers: Refabricated 1.1.1
		enchant_giver: Enchant Giver 1.0.3
		enchantment_lore: EnchantmentLore 1.1.1+MC1.16.1-1.16.5
		enchantmentdescriptions: Enchantment Descriptions 2.0.1
		enhancedcelestials: Enhanced Celestials 1.0.2
		environmentz: EnvironmentZ 1.0.12
		extraalchemy: Extra Alchemy 1.4.2
		extraorigins: Extra Origins 1.16.5-10
		fabric: Fabric API 0.32.0+1.16
		fabric-api-base: Fabric API Base 0.2.1+9354966b7d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.0.0+dc716ea17d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.1+ca58154a7d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+ca58154a7d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.0+f9b3753b7d
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+ca58154a7d
		fabric-containers-v0: Fabric Containers (v0) 0.1.11+9354966b7d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.1+ca58154a7d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.3+ca58154a7d
		fabric-dimensions-v1: fabric-dimensions-v1 2.0.6+9354966b7d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.0.3+ca58154a7d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.2+ca58154a7d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+ca58154a7d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.6+ca58154a7d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.1+ca58154a7d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.3+ca58154a7d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+9354966b7d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.1+ca58154a7d
		fabric-language-kotlin: Fabric Language Kotlin 1.5.0+kotlin.1.4.31
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.1+ca58154a7d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.2+ca58154a7d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+ca58154a7d
		fabric-models-v0: Fabric Models (v0) 0.2.1+ca58154a7d
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.1+ca58154a7d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.8+ca58154a7d
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+ca58154a7d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.9.4+9354966b7d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.2+ca58154a7d
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+ca58154a7d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.4+ca58154a7d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.1+ca58154a7d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.4+ca58154a7d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.2.1+ca58154a7d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+ca58154a7d
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+ca58154a7d
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+ca58154a7d
		fabric-rendering-v1: Fabric Rendering (v1) 1.5.1+ca58154a7d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.2+ca58154a7d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.0+c045166c7d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.5+9354966b7d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.4+ca58154a7d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.1+ca58154a7d
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+ca58154a7d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.6+ca58154a7d
		fabricloader: Fabric Loader 0.11.3
		fadeinchunks: Fade In Chunks 1.0.0
		fading: Fading 1.2.1
		fallflyinglib: FallFlyingLib 1.1.0
		fasttransferlib: Fast Transfer Lib 0.3.1
		fiber: fiber 0.23.0-2
		firstperson: FirstpersonModel 2.0.0
		gbfabrictools: GBfabrictools 1.3.2+1.16.4
		gentlefawn: GentleFawn 1.1.4-MC1.16.5-fabric
		glassdoor: Glass Doors 1.5.0-1.16
		go-fish: Go Fish 1.2.1-beta-1.16.4
		haema: Haema 1.5.3
		illuminations: Illuminations 1.4.6
		improved-stations: Improved Stations 2.0.3
		indium: Indium 1.0.0
		inventorysorter: Inventory Sorter 1.7.8-1.16
		itemscroller: Item Scroller 0.15.0-dev.20210111.205029
		jankson: Jankson 3.0.1+j1.2.0
		java: Java HotSpot(TM) 64-Bit Server VM 8
		javassist_javassist: javassist 3.12.1.GA
		kirin: Kirin UI 1.7-1.16.2
		kyrptconfig: Kytpt Config 1.1.6-1.16
		lambdynlights: LambDynamicLights 1.3.4+1.16
		libblockattributes: LibBlockAttributes 0.8.4
		libblockattributes_core: LibBlockAttributes (Core) 0.8.7
		libblockattributes_fluids: LibBlockAttributes (Fluids) 0.8.7
		libblockattributes_items: LibBlockAttributes (Items) 0.8.7
		libcd: LibCapableData 3.0.3+1.16.3
		libgui: LibGui 3.3.5+1.16.5
		libstructure: LibStructure 1.5
		lifts: Lifts 1.0.4-BETA
		magic_lanterns: Magic Lanterns 1.4.5
		magna: Magna 1.5.1-1.16.5
		malilib: MaLiLib 0.10.0-dev.21+arne.2
		mambience: MAmbience 3.1.0+1.16.5
		maybe-data: Maybe data 1.0.0-1.16.2
		mcda: MC Dungeons Armors 1.3.3
		mcdw: MC Dungeons Weapons 3.0.5
		mechanix: Mechanix 3.3.0
		medievalweapons: Medieval Weapons 1.0.8
		megane: megane 3.1.3+1.16.5
		megane-api: megane-api 3.1.3+1.16.5
		megane-impl: megane-impl 3.1.3+1.16.5
		megane-runtime: megane-runtime 3.1.3+1.16.5
		minecraft: Minecraft 1.16.5
		moborigins: Mob Origins 1.3.1
		mobz: MobZ 2.4.6
		modern_industrialization: Modern Industrialization 0.3.15
		modmenu: Mod Menu 1.16.8
		mostructures: Mo' Structures 1.1.0+mc.1.16.5
		net_dv8tion_jda: JDA 4.2.0_227
		net_sf_trove4j_trove4j: trove4j 3.0.3
		notenoughanimations: NotEnoughAnimations 1.1.0
		notenoughcrashes: Not Enough Crashes 3.1.8
		okzoomer: Ok Zoomer 4.0.1+1.16.2
		org_apache_commons_commons-collections4: commons-collections4 4.1
		org_aperlambda_lambdajcommon: lambdajcommon 1.8.1
		org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.4.31
		org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.4.31
		org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.4.31
		org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.4.31
		org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.4.2
		org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.4.2
		org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.1.0
		org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.1.0
		org_json_json: json 20170516
		org_slf4j_slf4j-api: slf4j-api 1.7.25
		org_slf4j_slf4j-nop: slf4j-nop 1.7.25
		origins: Origins 0.6.1
		origins-classes: Origins: Classes 1.1.1
		particlerain: Particle Rain 1.2.0
		patchouli: Patchouli 1.16.4-48-FABRIC
		pehkui: Pehkui 1.9.0+21w10a
		phosphor: Phosphor 0.7.1+build.11
		polymorph: Polymorph 0.0.10-1.16.5
		presencefootsteps: Presence Footsteps r29+1.16.4
		reach-entity-attributes: Reach Entity Attribute 1.0.1
		reborncore: Reborn Core 4.7.2+build.132
		repurposed_structures: Repurposed Structures 1.16.5-1.9.0
		roughlyenoughitems: Roughly Enough Items 5.10.184
		roughlyenoughitems-api: REI (API) 5.10.184
		roughlyenoughitems-default-plugin: REI (Default Plugin) 5.10.184
		roughlyenoughitems-runtime: REI (Runtime) 5.10.184
		roughlyenoughresources: Roughly Enough Resources 2.1.3
		rpg-hud: RPG-Hud 3.8.1
		rpgstats: RPGStats 3.2.3+1.16.5
		rpgz: RPGZ 0.4.6
		satin: Satin 1.6.0-nightly.20w48a
		seasons: Fabric Seasons 1.1.1-BETA
		shulkerutils: shulkerutils 1.0.1-1.16
		smoothboot: Smooth Boot 1.16.5-1.5.0
		sodium: Sodium 0.1.1-SNAPSHOT
		somnus: Somnus API 0.0.10-1.16.4
		spinnery: Spinnery 3.2.1
		spruceui: SpruceUI 2.0.4+1.16
		stacc: Stacc 1.0.1
		static-content: Static Content 1.0.1-1.16.2
		staticdata: Static Data 1.1.2
		step-height-entity-attribute: Step Height Entity Attribute 1.0.0
		structure-helpers: Structure Helpers 3.0.0
		team_reborn_energy: Energy 0.1.1
		techreborn: Tech Reborn 3.8.1+build.211
		terraform-biome-builder-api-v1: Terraform Biome Builder API (v1) 1.0.0+build.4
		terraform-shapes-api-v1: Terraform Shapes API (v1) 1.0.0+build.4
		terraform-tree-api-v1: Terraform Tree API (v1) 1.0.1+build.4
		terraform-wood-api-v1: Terraform Wood API (v1) 1.0.0+build.4
		thats-my-chest: That's My Chest! 1.1.0
		things: Things 0.1
		tk_valoeghese_zoesteriaconfig: ZoesteriaConfig 1.3.2
		transliterationlib: TRansliterationLib 1.0.2
		trinkets: Trinkets 2.6.7
		upgradedshulkers: Upgraded Shulkers 1.0.0-1.16
		vanguard: Vanguard 1.0.5
		vanilla-hammers: Vanilla Hammers 3.0.2-1.16.5
		varied-mobs: Varied Mobs 1.0.0
		voicechat: Simple Voice Chat 1.16.5-1.1.1
		voxelmap: VoxelMap 1.10.15
		voyager: Voyager 1.0.0
		waila: Hwyla 1.9.22
		waystones: Waystones 1.1.1.1
		wildfire_gender: Wildfire's Gender Mod 1.16.4-1.0.0
		wildworld: Wild World 2.0.0+1.16.5
		xlpackets: XLPackets 1.1
		yosbr: YOSBR 0.1.1
		yungsapi: YUNG's API 1.16.5-Fabric-2
	Patchouli open book context: n/a
	Launched Version: fabric-loader-0.11.3-1.16.5
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: GeForce GTX 1050 Ti/PCIe/SSE2 GL version 4.6.0 NVIDIA 461.92, NVIDIA Corporation
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Resource Packs: Fabric Mods
	Current Language: English (US)
	CPU: 12x Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
	Client Crashes Since Restart: 2
	Integrated Server Crashes Since Restart: 0
```

**Comment by apace100** [comment]
Fixed in 0.6.2
